### PR TITLE
Make actions use `target=this` implicitly

### PR DESCRIPTION
**Pull request by zackthehuman**
This addresses the bug in #14469. Might need a new test. Please review @rwjblue @chancancode.

### CODE REVIEW

**Comment by zackthehuman** [comment]
I added a test.

File: packages/ember-template-compiler/lib/plugins/transform-action-syntax.js [hunk 1]
```diff
@@ -15,11 +15,13 @@
   with
 
   ```handlebars
- <button {{action this 'foo'}}>
- <button onblur={{action this 'foo'}}>
- <button onblur={{action this (action this 'foo') 'bar'}}>
+ <button {{action target=this 'foo'}}>
```
Comment (rwjblue) [inline packages/ember-template-compiler/lib/plugins/transform-action-syntax.js]: This isn't quite right, the hash has to be after positional args...

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: packages/ember-template-compiler/lib/plugins/transform-action-syntax.js [hunk 2]
```diff
@@ -15,11 +15,13 @@
...
+ <button onblur={{action target=this 'foo'}}>
+ <button onblur={{action target=this (action target=this 'foo') 'bar'}}>
```
Comment (sukima) [inline packages/ember-template-compiler/lib/plugins/transform-action-syntax.js]: Isn't this also backwards.

File: packages/ember-template-compiler/lib/plugins/transform-action-syntax.js [hunk 3]
```diff
@@ -62,6 +67,24 @@ function isAction(node) {
...
+function hashPairForKey(hash, key) {
+  for (let i = 0; i < hash.pairs.length; i++) {
+    let pair = hash.pairs[i];
+    if (pair.key === key) {
+      return pair;
+    }
+  }
+
+  return false;
+}
```
Comment (sukima) [inline packages/ember-template-compiler/lib/plugins/transform-action-syntax.js]: This function is unfortunately repeated. @rwjblue should the util function be DRY somewhere? And if so where?

File: packages/ember-template-compiler/lib/plugins/transform-action-syntax.js [hunk 4]
```diff
@@ -62,6 +64,20 @@ function isAction(node) {
...
+  }
+}
+
+function hashPairForKey(hash, key) {
+  for (let i = 0; i < hash.pairs.length; i++) {
+    let pair = hash.pairs[i];
+    if (pair.key === key) {
+      return pair;
+    }
+  }
```
Comment (sukima) [inline packages/ember-template-compiler/lib/plugins/transform-action-syntax.js]: This function is unfortunately repeated. @rwjblue should the util function be DRY somewhere? And if so where?

Comment (zackthehuman) [inline packages/ember-template-compiler/lib/plugins/transform-action-syntax.js]: I will extract it and then make another PR to update other refs.

File: packages/ember-template-compiler/lib/plugins/transform-action-syntax.js (hunk 2 above)
Comment (zackthehuman) [inline packages/ember-template-compiler/lib/plugins/transform-action-syntax.js]: Yes; it has been corrected.

**Comment by Serabe** [comment]
@zackthehuman is this ready for a new review? Thank you!

**Comment by zackthehuman** [comment]
@Serabe I think we may punt on this due to what was discussed in #14469. Closing for now.

**Comment by Serabe** [comment]
@zackthehuman thank you!

**Pull request closed without merging**